plugins/modules/tanzu_mission_control_secret.py

```diff
@@ -0,0 +1,793 @@
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.six.moves.urllib.parse import urlencode
+from ansible.module_utils.six.moves.urllib.error import HTTPError
+from ansible.module_utils.urls import open_url
+
+
+class TMCSecret(object):
+    """Represents a generic secret (either Cluster or ClusterGroup) in TMC."""
+
+    def __init__(self, secret_name, secret_type, cluster_namespace):
+        self._secret_name = secret_name
+        self._secret_type = (
+            "SECRET_TYPE_OPAQUE"
+            if secret_type == "opaque"
+            else "SECRET_TYPE_DOCKERCONFIGJSON"
+        )
+        self._cluster_namespace = cluster_namespace
+
+    def exists(self, api_host, access_token):
+        raise NotImplementedError()
+
+    def is_exported(self, api_host, access_token):
+        raise NotImplementedError()
+
+    def create(self, api_host, access_token, payload):
+        raise NotImplementedError()
+
+    def update(self, api_host, access_token, payload):
+        raise NotImplementedError()
+
+    def delete(self, api_host, access_token):
+        raise NotImplementedError()
+
+    def create_export(self, api_host, access_token):
+        raise NotImplementedError()
+
+    def delete_export(self, api_host, access_token):
+        raise NotImplementedError()
+
+
+class TMCClusterGroupSecret(TMCSecret):
+    """Represents a ClusterGroup secret in TMC."""
+
+    def __init__(self, cluster_group, cluster_namespace, secret_name, secret_type):
+        super().__init__(secret_name, secret_type, cluster_namespace)
+        self._cluster_group = cluster_group
+
+    def _generate_query_params(self):
+        return urlencode({"fullName.namespaceName": self._cluster_namespace})
+
+    def _generate_url(self, api_host):
+        return "https://{}/v1alpha1/clustergroups/{}/namespace/secrets".format(
+            api_host, self._cluster_group
+        )
+
+    def _get_current_spec(self, api_host, access_token):
+        url = "{}/{}?{}".format(
+            self._generate_url(api_host),
+            self._secret_name,
+            self._generate_query_params(),
+        )
+        response = open_url(
+            url,
+            headers={"Authorization": "Bearer {}".format(access_token)},
+        )
+
+        return json.loads(response.read())
+
+    def _generate_export_url(self, api_host, operation="get"):
+        if operation == "get":
+            return "https://{}/v1alpha1/clustergroups/{}/namespaces/{}/secretexports".format(
+                api_host, self._cluster_namespace, self._cluster_group
+            )
+        else:
+            return (
+                "https://{}/v1alpha1/clustergroups/{}/namespace/secretexports".format(
+                    api_host, self._cluster_group
+                )
+            )
+
+    def _prepare_request(self, payload):
+        new_payload = {}
+        for key, value in payload.items():
+            new_payload[key] = b64encode(value.encode()).decode()
```

Comment:
`encode()` and `decode()` might not work correctly on Python 2.7, since strings work differently there. Use `to_text()`, `to_binary()`, and `to_native()`, like
```suggestion
            new_payload[key] = to_native(b64encode(to_binary(value)))
```
Same in other places.

(You'll need `from ansible.module_utils.common.text.converters import to_bytes, to_native, to_text` resp. a subset of these.)